docs/typescript/testing.md

```diff
@@ -2,92 +2,246 @@
 
-  it('retries one failure', async function () {
-    // Make the first request fail, but subsequent requests succeed
-    let numCalls = 0;
-    sinon.stub(axios, 'get').callsFake(() => {
-      if (numCalls++ === 0) {
-        return Promise.reject(new Error('first error'));
-      }
-      return Promise.resolve({ data: { args: { answer: '88' } } });
-    });
-
-    const result = await client.execute(example);
-    assert.equal(result, 'The answer is 88');
+### Example setup
+
+> NOTE: `beforeAll` and `afterAll` are injected by `jest`.
+
+```ts
+import { TestWorkflowEnvironment } from '@temporalio/testing';
+import { Worker } from '@temporalio/worker';
+import { v4 as uuid4 } from 'uuid';
+import { httpWorkflow } from './workflows';
+import type * as Activities from './activities'; // Uses types to ensure our mock signatures match
+
+let testEnv: TestWorkflowEnvironment;
+
+beforeAll(async () => {
+  testEnv = await TestWorkflowEnvironment.create();
+});
+
+afterAll(async () => {
+  await testEnv?.teardown();
+});
+```
+
+### Mocking Activities
+
+Since the `TestWorkflowEnvironment` is meant for testing Workflows, you'd typically want to mock your activities in tests to avoid generating side-effects.
+
+```ts
+test('httpWorkflow with mock activity', async () => {
+  const { workflowClient, nativeConnection } = testEnv;
+
+  // Implement only the relevant activities for this workflow
+  const activities: Partial<typeof Activities> = {
+    makeHTTPRequest: async () => '99',
+  };
+  const worker = await Worker.create({
+    connection: nativeConnection,
+    taskQueue: 'test',
+    workflowsPath: require.resolve('./workflows'),
+    activities,
   });
+  const result = await worker.runUntil(
+    await workflowClient.execute(httpWorkflow, {
+      workflowId: uuid4(),
+      taskQueue: 'test',
+    })
+  );
+  expect(result).toEqual('The answer is 99');
+});
+```
+
+### Time skipping
+
+The built-in test server automatically "skips" (fast forwards) time when no activities are executing.
+If a Workflow sleeps for days, running it in the test environment will cause it to complete almost immediately.
 
-  it('bubbles up activity errors', async function () {
-    sinon
-      .stub(axios, 'get')
-      .callsFake(() => Promise.reject(new Error('example error')));
-
-    const err = await client.execute(example, { taskQueue: 'testhttp' }).then(
-      () => null,
-      (err) => err
-    );
-    assert.equal(err.name, 'WorkflowFailedError');
-    assert.equal(err.cause.cause.message, 'example error');
+<details>
+<summary>
+Workflow implementation
+</summary>
+
+<!--SNIPSTART typescript-timer-reminder-workflow-->
+<!--SNIPEND-->
+
+</details>
+
+```ts
+test('countdownWorkflow sends reminder email if processing does not complete in time', async () => {
```

Comment:
use snippet?

Reply:
I changed it a bit, need to use snippets for everything here eventually.